### ISSUE TITLE
PHP name or alias is not honored using phpbrew v2

### ISSUE THREAD

**Issue by ioweb-gr** [phpbrew/phpbrew#1305]
The php name is not honored if I use the syntax with "as" or with --name

### Output

root@d5server:~# phpbrew list
PHP Warning:  Module 'blackfire' already loaded in Unknown on line 0
* (system)
  php-8.1.12
  phpbrew7_4_32_fdev
  phpbrew7_0_33_fdev

### Expected Result

The installed version has the following name
```
phpbrew8_1_12_fdev
```

### Command

Paste your phpbrew install command here (be sure to install with `--debug` flag, e.g. `phpbrew --debug install`)

export PHPBREW_VERSION='8.1.12'
export PHPBREW_VERSION_ALIAS='phpbrew8_1_12_fdev'

phpbrew --debug install "$PHPBREW_VERSION" +default +small +mb +dbs +fpm -wddx +openssl=/usr -- --with-pspell --disable-shmop as "$PHPBREW_VERSION_ALIAS"

or 

phpbrew --debug install "$PHPBREW_VERSION" +default +small +mb +dbs +fpm -wddx +openssl=/usr -- --with-pspell --disable-shmop --name="$PHPBREW_VERSION_ALIAS"

### Platform

root@d5server:~# phpbrew --version
PHP Warning:  Module 'blackfire' already loaded in Unknown on line 0
phpbrew - 2.0.0
cliframework core: 2.5.4

root@d5server:~# cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.6 LTS"

Running PHP: 

```
root@d5server:~# php -v
PHP Warning:  Module 'blackfire' already loaded in Unknown on line 0
PHP 7.2.24-0ubuntu0.18.04.15 (cli) (built: Nov  2 2022 09:09:52) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.24-0ubuntu0.18.04.15, Copyright (c) 1999-2018, by Zend Technologies
    with the ionCube PHP Loader + ionCube24 v10.4.5, Copyright (c) 2002-2020, by ionCube Ltd.
    with blackfire v1.86.1~linux-x64-non_zts72, https://blackfire.io, by Blackfire
```

Installing PHP:
8.1.12

**Comment by peter279k** [comment]
It seems that the issue is related to [this](https://github.com/phpbrew/phpbrew/issues/871).

**Comment by peter279k** [comment]
After checking the issue, it seems the configure script about aliasing option is changed.

**Comment by peter279k** [comment]
After running the `PHPBrew` version command for alias feature in the `Ubuntu 18.04`, I don't get above issue.

Here are commands I run:

```sh
root@ubuntu-s-4vcpu-8gb-amd-sgp1-01:~# export PHPBREW_VERSION='8.1.12'
root@ubuntu-s-4vcpu-8gb-amd-sgp1-01:~# export PHPBREW_VERSION_ALIAS='phpbrew8_1_12_fdev'
root@ubuntu-s-4vcpu-8gb-amd-sgp1-01:~# phpbrew --debug install "$PHPBREW_VERSION" +default +small +mb +dbs +fpm -ipc -wddx +openssl=/usr as "$PHPBREW_VERSION_ALIAS"
*WARNING* You're running phpbrew as root/sudo. Unless you're going to install system-wide phpbrew, this might cause problems.
===> phpbrew will now build 8.1.12
......
Congratulations! Now you have PHP with 8.1.12 as phpbrew8_1_12_fdev

* To configure your installed PHP further, you can edit the config file(s) at
    /root/.phpbrew/php/phpbrew8_1_12_fdev/etc/cli/php.ini

To use the newly built PHP, try the line(s) below:

    $ phpbrew use phpbrew8_1_12_fdev

Or you can use switch command to switch your default php to phpbrew8_1_12_fdev:

    $ phpbrew switch phpbrew8_1_12_fdev
```

To disable the `shmop` extension, it can use the `-ipc` variant.

To enable/install the `pspell`, it can run the following command after building the current PHP version is completed:

```sh
$ phpbrew use phpbrew8_1_12_fdev
$ phpbrew ext install pspell
```

BTW, it can use the following command for `PHP 7.x` building:

```
phpbrew --debug install "7.4" as "7.4-dev" +default +small +mb +dbs +fpm -ipc -wddx +openssl=/usr
```

**Comment by ioweb-gr** [comment]
The only difference in this case appears to be that you have no positional arguments after enabling / disabling the variants . Can you verify that with the original command you get the same issue as I do ?

**Comment by peter279k** [comment]
The original command I cannot run because you set the wrong alias argument inside the `phpbrew install` command.

The alias is for the PHPBrew command and it's not for the `./configure` script. And it should change into following commands:

```sh
$ export PHPBREW_VERSION='8.1.12'
$ export PHPBREW_VERSION_ALIAS='phpbrew8_1_12_fdev'
$ phpbrew --debug install "$PHPBREW_VERSION" as "$PHPBREW_VERSION_ALIAS" +default +small +mb +dbs +fpm -wddx +openssl=/usr -- --with-pspell --disable-shmop
```

**Comment by ioweb-gr** [comment]
Just to clarify in the past this command was working flawlessly I have installations with 

phpbrew - 1.27.0
cliframework core: 2.5.4

Where I've installed at least php 7.4 and 7.2 with it. 

~$ phpbrew list
* phpbrew7_4_28_fdev
  phpbrew7_2_34_fdev

Hence the confusion. 

I'm glad it's just this so I can work through it but your original comment doesn't apply? 

>After checking the issue, it seems the configure script in PHP 8.x about aliasing option is changed.

Also I got a weird error now 

To use the newly built PHP, try the line(s) below:

    $ phpbrew use phpbrew8_1_12_fdev

Or you can use switch command to switch your default php to phpbrew8_1_12_fdev:

    $ phpbrew switch phpbrew8_1_12_fdev

Enjoy!
root@d5server:~# phpbrew use phpbrew8_1_12_fdev
PHP Warning:  Module 'blackfire' already loaded in Unknown on line 0
Exception: You should not see this. If you see this, it means you didn't load the ~/.phpbrew/bashrc script. Please check if bashrc is sourced in your shell.
Trace:

    0) PhpBrew\Command\VirtualCommand->execute('phpbrew8_1_12_fdev')
    1) call_user_func_array([PhpBrew\Command\UseCommand, 'execute'], ['phpbrew8_1_12_fdev'])
    2) CLIFramework\CommandBase->executeWrapper(['phpbrew8_1_12_fdev'])
    3) CLIFramework\Application->run(['/usr/local/bin/phpbrew', 'use', 'phpbrew8_1_12_fdev'])
    4) PhpBrew\Console->runWithTry(['/usr/local/bin/phpbrew', 'use', 'phpbrew8_1_12_fdev'])
    5) require('phar:///usr/local/bin/phpbrew/bin/phpbrew')

root@d5server:~# . ~/.phpbrew/bashrc
root@d5server:~# phpbrew use phpbrew8_1_12_fdev
PHP Fatal error:  Uncaught TypeError: Return value of HumbugBox402\KevinGH\RequirementChecker\RequirementCollection::add() must be an instance of HumbugBox402\KevinGH\RequirementChecker\void, none returned in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php:25
Stack trace:
#0 phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php(28): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->add(Object(HumbugBox402\KevinGH\RequirementChecker\Requirement))
#1 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(81): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->addRequirement(Object(HumbugBox402\KevinGH\RequirementChecker\IsPhpVersionFulfilled), 'The application...', 'The application...')
#2 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::retrieveRequirements()
#3 phar:///usr/local/bin/phpbrew/.box/bin/check-requirements.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::checkRequirements()
#4 /usr/local/bin/phpbrew(12) in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php on line 25


And now I lost all my php versions also when I load my shell again.

**Comment by ioweb-gr** [comment]
So my previous phpbrew installation was using 

  phpbrew8_1_12_fdev
  phpbrew7_4_32_fdev
*  phpbrew7_0_33_fdev

After upgrading, and installing the new phpbrew version, I can no longer switch to phpbrew7_0_33_fdev because everything breaks. 

It seems because compatibility cannot be retained, it's hard to juggle threw different php versions 

What's the recommended procedure for maintaining both php < 7.2 and > 8.0

**Comment by peter279k** [comment]
Could you run the following steps?

- Using the latest PHPBrew 1.x version.
- Upgrading the PHPBrew 1.x to the 2.x version.
- Running the `phpbrew init` again.
- Running the `phpbrew list` and `phpbrew use phpbrew7_0_33_fdev` commands.

**Comment by peter279k** [comment]
I try to reproduce above [issue](https://github.com/phpbrew/phpbrew/issues/1305#issuecomment-1365893947) and here are commands I execute:

```sh
$ wget https://github.com/phpbrew/phpbrew/releases/download/1.27.0/phpbrew.phar
$ mv phpbrew.phar /usr/local/bin/phpbrew
$ chmod +x /usr/local/bin/phpbrew
$ phpbrew init
$ echo "[[ -e ~/.phpbrew/bashrc ]] && source ~/.phpbrew/bashrc" >> ~/.bashrc
$ phpbrew --debug install --stdout 7.0 as 7.0-dev +default
$ phpbrew off

$ wget https://github.com/phpbrew/phpbrew/releases/download/2.0.0/phpbrew.phar
$ mv phpbrew.phar /usr/local/bin/phpbrew
$ chmod +x /usr/local/bin/phpbrew
$ phpbrew list
* (system)
  7.0-dev
$ phpbrew use 7.0-dev
$ php -v
PHP 7.0.33 (cli) (built: Dec 28 2022 03:29:44) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
......
$ phpbrew list
PHP Fatal error:  Uncaught TypeError: Return value of HumbugBox402\KevinGH\RequirementChecker\RequirementCollection::add() must be an instance of HumbugBox402\KevinGH\RequirementChecker\void, none returned in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php:25
Stack trace:
#0 phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php(28): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->add(Object(HumbugBox402\KevinGH\RequirementChecker\Requirement))
#1 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(81): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->addRequirement(Object(HumbugBox402\KevinGH\RequirementChecker\IsPhpVersionFulfilled), 'The application...', 'The application...')
#2 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::retrieveRequirements()
#3 phar:///usr/local/bin/phpbrew/.box/bin/check-requirements.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::checkRequirements()
#4 /usr/local/bin/phpbrew(12) in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php on line 25

Fatal error: Uncaught TypeError: Return value of HumbugBox402\KevinGH\RequirementChecker\RequirementCollection::add() must be an instance of HumbugBox402\KevinGH\RequirementChecker\void, none returned in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php:25
Stack trace:
#0 phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php(28): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->add(Object(HumbugBox402\KevinGH\RequirementChecker\Requirement))
#1 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(81): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->addRequirement(Object(HumbugBox402\KevinGH\RequirementChecker\IsPhpVersionFulfilled), 'The application...', 'The application...')
#2 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::retrieveRequirements()
#3 phar:///usr/local/bin/phpbrew/.box/bin/check-requirements.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::checkRequirements()
#4 /usr/local/bin/phpbrew(12) in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php on line 25
root@ubuntu-s-8vcpu-16gb-amd-sgp1-01:~# phpbrew switch 7.0-dev
PHP Fatal error:  Uncaught TypeError: Return value of HumbugBox402\KevinGH\RequirementChecker\RequirementCollection::add() must be an instance of HumbugBox402\KevinGH\RequirementChecker\void, none returned in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php:25
Stack trace:
#0 phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php(28): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->add(Object(HumbugBox402\KevinGH\RequirementChecker\Requirement))
#1 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(81): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->addRequirement(Object(HumbugBox402\KevinGH\RequirementChecker\IsPhpVersionFulfilled), 'The application...', 'The application...')
#2 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::retrieveRequirements()
#3 phar:///usr/local/bin/phpbrew/.box/bin/check-requirements.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::checkRequirements()
#4 /usr/local/bin/phpbrew(12) in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php on line 25

Fatal error: Uncaught TypeError: Return value of HumbugBox402\KevinGH\RequirementChecker\RequirementCollection::add() must be an instance of HumbugBox402\KevinGH\RequirementChecker\void, none returned in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php:25
Stack trace:
#0 phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php(28): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->add(Object(HumbugBox402\KevinGH\RequirementChecker\Requirement))
#1 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(81): HumbugBox402\KevinGH\RequirementChecker\RequirementCollection->addRequirement(Object(HumbugBox402\KevinGH\RequirementChecker\IsPhpVersionFulfilled), 'The application...', 'The application...')
#2 phar:///usr/local/bin/phpbrew/.box/src/Checker.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::retrieveRequirements()
#3 phar:///usr/local/bin/phpbrew/.box/bin/check-requirements.php(10): HumbugBox402\KevinGH\RequirementChecker\Checker::checkRequirements()
#4 /usr/local/bin/phpbrew(12) in phar:///usr/local/bin/phpbrew/.box/src/RequirementCollection.php on line 25
```

It seems that the issue is about the `HumbugBox`. @theofidry, could you check the issue?

**Comment by theofidry** [comment]
@peter279k to double check with version of Box is used, but that was a change in [4.0.0](https://github.com/box-project/box/releases/tag/4.0.0) where the requirement checker now uses 7.2.4+ (and drops anything lower). The rational is explained in https://github.com/box-project/box/pull/674.

To execute the code nonetheless you can use `BOX_REQUIREMENT_CHECKER=0`.

**Comment by peter279k** [comment]
I think the issue is resolved in the `2.1.0` version. Here are the following commands I run:

```bash
$ phpbrew --version
phpbrew - 2.1.0
cliframework core: 2.5.4
$ phpbrew list
* (system)
  php-8.1.14
  php-8.1.13
  php-7.4.33
  phpbrew7_2_34_fdev
$ phpbrew use phpbrew7_2_34_fdev
$ php -v
PHP 7.2.34 (cli) (built: Apr 14 2023 18:39:43) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
$ phpbrew list
  php-8.1.14
  php-8.1.13
  php-7.4.33
* phpbrew7_2_34_fdev
```
